lib/msgraph/client.go

```diff
@@ -198,6 +223,29 @@ func (c *Client) request(ctx context.Context, method string, uri string, header
 			}
 		}
 
+		requestID := uuid.NewString()
+		// https://learn.microsoft.com/en-us/graph/best-practices-concept#reliability-and-support
+		req.Header.Set("client-request-id", requestID)
+
+		{
+			authz := req.Header.Get("Authorization")
+			req.Header.Set("Authorization", "redacted")
+
+			// Unescape the query for clearer logs.
+			u := uri
+			if val, err := url.QueryUnescape(uri); err == nil {
+				u = val
+			}
+
+			c.logger.Log(req.Context(), log.TraceLevel, "Executing HTTP request",
+				"url", u,
+				"header", req.Header,
+				"retry_count", retryCount,
+				"clock_time", c.clock.Now(),
+			)
+			req.Header.Set("Authorization", authz)
+		}
```

Comment:
I am not sure this needs to go into production code. 

In the PR description, you mentioned that it was helpful during test. 
For tests, we can just dump requests inside the [test http handler](https://github.com/gravitational/teleport/blob/1970fb19dae8242257af3e7af389687f48693b12/lib/msgraph/client_test.go#L174C1-L178C4) 
https://pkg.go.dev/net/http/httputil#DumpRequest

Reply:
Removed in 624722668dcf54a278437681a3b06fe153366525 and added request dumping in gravitational/teleport.e@411724dfb1718bdcc7d98.